addons/website/static/src/components/configurator/configurator.js

```diff
@@ -0,0 +1,519 @@
+    {name: 'CONFIGURATOR_WELCOME_SCREEN_LANG', path: '/{{lang}}/website/configurator/1', component: WelcomeScreen},
+];
+
+//---------------------------------------------------------
+// Store
+//---------------------------------------------------------
+
+const getters = {
+    getWebsiteTypes() {
+        return Object.values(WEBSITE_TYPES);
+    },
+
+    getSelectedType(_, id) {
+        return id ? WEBSITE_TYPES[id] : undefined;
+    },
+
+    getWebsitePurpose() {
+        return Object.values(WEBSITE_PURPOSES);
+    },
+
+    getSelectedPurpose(_, id) {
+        return id ? WEBSITE_PURPOSES[id] : undefined;
+    },
+
+    getFeatures({state}) {
+        return Object.values(state.features);
+    },
+
+    getPalettes({state}) {
+        return Object.values(state.palettes);
+    },
+
+    getThemeName({state}, idx) {
+        return state.themes.length > idx ? state.themes[idx].name : undefined;
+    }
+};
+
+const actions = {
+    selectWebsiteType({state}, id) {
+        Object.values(state.features).forEach((feature) => {
+            feature.selected = feature.website_types_preselection.includes(WEBSITE_TYPES[id].name);
+        });
+        state.selectedType = id;
+    },
+    selectWebsitePurpose({state}, id) {
+        state.selectedPurpose = id;
+    },
+    selectIndustry({state}, name) {
+        state.selectedIndustry = name;
+    },
+    changeLogo({state}, data) {
+        state.logo = data;
+    },
+    selectPalette({state}, paletteId) {
+        state.selectedPalette = state.allPalettes[paletteId];
+    },
+    toggleFeature({state}, featureId) {
+        const feature = state.features[featureId];
+        const websiteType = WEBSITE_TYPES[state.selectedType];
+        const forceFeatureActive = websiteType && feature.website_types_preselection.includes(websiteType.name);
+        feature.selected = !feature.selected || forceFeatureActive;
+    },
+    setRecommendedPalette({state}, color1, color2) {
+        let palettes = [];
+        if (color1 && color2) {
+            Object.values(state.allPalettes).forEach(palette => {
+                const delta1 = deltaE(color1, palette.color1);
+                const delta2 = deltaE(color2, palette.color2);
+                palette.score = (delta1 + delta2) / 2;
+            });
+            palettes = Object.values(state.allPalettes).sort((a, b) => a.score - b.score);
+            state.recommendedPalette = palettes[0];
+        } else {
+            palettes = Object.values(state.allPalettes);
+        }
+        const selectedPalettes = {};
+        palettes.slice(1, MAX_PALETTES + 1).forEach((palette) => {
+            selectedPalettes[palette.id] = palette;
+        });
+        state.palettes = selectedPalettes;
+    },
+    updateRecommendedThemes({state}, themes) {
+        state.themes = themes.slice(0, 3);
+    }
+};
+
+async function getInitialState() {
+
+    var results = {};
```

Comment:
Why not just define that instead of the `try` block? Also initial value doesn't seem useful.